drivers/sensor/mockup/battery/battery.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <kernel.h>
+#include <sensor.h>
+
+struct battery_data {
+	u8_t battery_lvl;
+};
+
+static struct device *device;
+static int battery_init(struct device *dev);
+static sensor_trigger_handler_t trigger_callback;
```

Comment:
Don't use static variables for the device pointer and trigger_callback, as this will make your driver difficult to instantiate more than once. Move the variables into `struct battery_data` instead and use the `CONTAINER_OF` macro.
https://github.com/zephyrproject-rtos/zephyr/blob/148c7dd8a3fb989e3358425023809f74b71da936/drivers/sensor/fxos8700/fxos8700_trigger.c#L130-L136